code/mob.dm

```diff
@@ -486,10 +489,9 @@
 	if (illumplane) //Wire: Fix for Cannot modify null.alpha
 		illumplane.alpha = 255
 
-	if(HAS_MOB_PROPERTY(src, PROP_PROTANOPIA))
-		// creating a local var for this is actually necessary, byond freaks the fuck out if you do `color = list(blahblah)`. Why? I wish I knew.
-		var/list/matrix_protanopia = list(MATRIX_PROTANOPIA)
-		src.client?.color = matrix_protanopia
+	src.client?.color = src.active_color_matrix
```

Comment:
Seeing `?.` on the left side of an assignment operator is weird. I know this is from earlier code, but this feels very wrong to me.

Reply:
I think Pali suggested that originally, since not all mobs have a client, and it led to runtimes. Plus, since this relies on a client to work, and is applied to the mob's client on login, having it not do anything without a client present seems like it'd be okay.